src/base/bittorrent/torrentinfo.cpp

```diff
@@ -363,8 +363,11 @@ TorrentInfo::PieceRange TorrentInfo::filePieces(const int fileIndex) const
     const lt::file_storage &files = nativeInfo()->files();
     const auto fileSize = files.file_size(LTFileIndex {fileIndex});
     const auto fileOffset = files.file_offset(LTFileIndex {fileIndex});
-    return makeInterval(static_cast<int>(fileOffset / pieceLength()),
-                        static_cast<int>((fileOffset + fileSize - 1) / pieceLength()));
+
+    const int begin = static_cast<int>(fileOffset / pieceLength());
+    const int end = (fileSize > 0) ? static_cast<int>((fileOffset + fileSize - 1) / pieceLength())
+                                   : begin;
```

Comment:
Wait... If file has size of 0 then it has no pieces, isn't it? So you need to return some kind of invalid PieceRange. Or am I wrong?

Reply:
Just tested, for a file with `fileSize == 0`, `fileOffset` is `0`.
so I think it should still return valid PieceRange.

Reply:
>Just tested, for a file with fileSize == 0, fileOffset is 0.

But it still has no pieces, isn't it? If you return [0; 0] range it means the file has 1 piece. Seems incorrect to me...

Reply:
I updated this part, these code are convoluted...

Reply:
Sorry, but you didn't fix zero-sized files problem.
>If you return [0; 0] range it means the file has 1 piece. Seems incorrect to me...

It still here.
Apparently, the problem is a little more complicated than it seems...